v2/client/client.go

```diff
@@ -29,28 +29,17 @@ type Client struct {
 // NewClient creates a new Client set to connect to the passed
 // hostname.
 func NewClient(voucherURL string) (*Client, error) {
-	if "" == voucherURL {
-		return nil, errNoHost
-	}
-
-	u, err := url.Parse(voucherURL)
-	if nil != err {
-		return nil, fmt.Errorf("could not parse voucher hostname: %s", err)
-	}
-	if "" == u.Scheme {
-		u.Scheme = "https"
-	}
+	return newClient(voucherURL, &http.Client{})
+}
 
+// NewAuthClient creates a new auth Client set to connect to the passed
+// hostname using tokens.
+func NewAuthClient(voucherURL string) (*Client, error) {
 	authClient, err := idtoken.NewClient(context.Background(), voucherURL)
-	if nil != err {
-		authClient = &http.Client{}
-	}
-
-	client := &Client{
-		url:        u,
-		httpClient: authClient,
+	if err != nil {
+		return NewClient(voucherURL)
```

Comment:
why do we want to return a `client` here? If there's an error, the token won't be appended to the header. Is that what we want in this function? (based on the comment)

If I understand correctly, we want to return an error?

Reply:
I believe we default to basic-auth if that is the case?

correct me if I'm wrong @lynnsh?

Reply:
Yes, I thought to let it fail in the app that calls auth client, but this approach is clearer.

Reply:
@rxbchen I don't think there are specific steps we are supposed to follow 🤔 If it's on CloudRun side, it would not accept basic auth anyway, so I'm not sure there is a value in calling http client.

Reply:
But if I use the auth function, I'm expecting to use whatever is not the default (`NewClient` is already using basic-auth). In that case, an error would indicate that auth with the token didn't work as expected, so it might be clearer from a ux point of view

Reply:
Yes, that's why I changed it, I think it's more clear 👍